ui/hooks/useAsyncResult.ts

```diff
@@ -32,7 +32,9 @@ export function useAsyncResult<T>(
   });
 
   useEffect(() => {
-    setResult({ pending: true });
+    if (!result.pending) {
+      setResult({ pending: true });
```

Comment:
Minor, would it be more standard to track the `pending`, `error`, and `value` in distinct `useState` hooks to enable React to do this comparison for us and only re-render if the value changes?

We could also `useMemo` the result to ensure we don't bother generating a new object reference if any of the properties change?